kernel/work_q.c

```diff
@@ -132,12 +128,12 @@ int k_delayed_work_submit_to_queue(struct k_work_q *work_q,
 
 int k_delayed_work_cancel(struct k_delayed_work *work)
 {
-	if (!work->work_q) {
-		return -EINVAL;
-	}
-
 	k_spinlock_key_t key = k_spin_lock(&lock);
-	int ret = work_cancel(work);
+	int ret = -EINVAL;
+
+	if (work->work_q) {
```

Comment:
You need to explicitly compare pointer with NULL (misra 14.4)

Reply:
done

Reply:
@nashif please revisit, thanks.